### ISSUE TITLE
need to handle dropout/batch_norm operator in testing mode

### ISSUE THREAD

**Issue by QiJune** [PaddlePaddle/Paddle#5733]
Dropout and batch norm operator do different between training and testing mode. We should handle these operators when users inference data after training.

**Comment by qingqing01** [comment]
@QiJune I'm wondering whether the flag `is_test` is needed in most operators. Many operators have intermediate variables, which are usually used to store the intermediate result to used in gradient operators.  But for the inference, all these intermediate variables can be removed.

**Comment by QiJune** [comment]
@qingqing01 Yes, I agree with you. We need to do some optimization in inference. We now have `Prune` method to transform a ProgramDesc to another ProgramDesc. And inference ProgrameDesc will be generated with `is_test` flag. 
We need a detailed design of how to optimize inference. Handling Dropout and batch_norm operator is the first step.
